Refactoring GUI

TickyWill opened the pull request:
This version integrates deep modifications of the bmgui package to rationalize the tkinter widgets definition and placing in main tk window and related pages.
A bug in update_impact_factors module has been fixed.
 Thanks in advance for your review.

Commit:
Changes in globals names (still on going)

Commit:
Partial refactoring of GUI through globals rename and management of use

Commit:
Introduced a folder of Institute parameter located out of the working folder

Commit:
Updated dafault value of working folder

Commit:
Updated requirments.txt file

Commit:
Fixed a bug related to the type of the working folder path in main_page.py

Commit:
Refactored and consolidated bmfuncts.update_impact_factors module

Commit:
Continued refactoring of GUI modules

Commit:
Created a module of functions shared by all GUI pages

Commit:
Created a module of classes shared by pages and a modules of useful functions for main page

Commit:
Replaced repeated lines in GUI pages by shared functions of bmgui.pages_utils module

Commit:
Finalized GUI refactoring and docstring completion after pylint scan

BaptisteRefalo commented on bmfuncts/authors_analysis.py:
```diff
@@ -345,19 +345,19 @@ def authors_analysis(institute, org_tup, bibliometer_path, datatype,
     if_analysis_name = None
-    _ = save_final_results(institute, org_tup, bibliometer_path, datatype, corpus_year,
+    _ = save_final_results(institute, org_tup, wf_path, datatype, corpus_year,
                            if_analysis_name, results_to_save_dict, verbose=False)
```
The "if_analysis_name" parameter should be a string and here, it is set to None -> perhaps using the "None" string value instead of None on line 359 would be better ?

TickyWill replied:
if_analysis_name  set to "None"  everywhere needed. Thanks

BaptisteRefalo commented on bmfuncts/consolidate_pub_list.py:
```diff
@@ -207,22 +207,22 @@ def built_final_pub_list(institute, org_tup, bibliometer_path, datatype,
     if_analysis_name = None
-    final_save_message = save_final_results(institute, org_tup, bibliometer_path,
+    final_save_message = save_final_results(institute, org_tup, wf_path,
                                             datatype, corpus_year, if_analysis_name,
```
[save_final_results() function] The "if_analysis_name" parameter should be a string and here, it is set to None -> perhaps using the "None" string value instead of None on line 224 would be better ?

BaptisteRefalo commented on bmfuncts/coupling_analysis.py:
```diff
@@ -359,13 +359,13 @@ def coupling_analysis(institute, org_tup, bibliometer_path,
         if_analysis_name = None
-        _ = save_final_results(institute, org_tup, bibliometer_path, datatype, year,
+        _ = save_final_results(institute, org_tup, wf_path, datatype, year,
                                if_analysis_name, results_to_save_dict, verbose=False)
```
[save_final_results() function] The "if_analysis_name" parameter should be a string and here, it is set to None -> perhaps using the "None" string value instead of None on line 337 would be better ?

TickyWill replied:
Done. Thanks

BaptisteRefalo commented on bmfuncts/doctype_analysis.py:
```diff
@@ -484,24 +484,24 @@ def doctype_analysis(institute, org_tup, bibliometer_path, datatype,
     if_analysis_name = None
-    _ = save_final_results(institute, org_tup, bibliometer_path, datatype,
+    _ = save_final_results(institute, org_tup, wf_path, datatype,
                            corpus_year, if_analysis_name, results_to_save_dict,
```
[save_final_results() function] The "if_analysis_name" parameter should be a string and here, it is set to None -> perhaps using the "None" string value instead of None on line 503 would be better ?

TickyWill replied:
Done. Thanks.

BaptisteRefalo commented on bmfuncts/keywords_analysis.py:
```diff
@@ -235,11 +235,11 @@ def keywords_analysis(institute, org_tup, bibliometer_path, datatype,
     if_analysis_name = None
-    _ = save_final_results(institute, org_tup, bibliometer_path, datatype, year,
+    _ = save_final_results(institute, org_tup, wf_path, datatype, year,
                            if_analysis_name, results_to_save_dict, verbose=False)
```
[save_final_results() function] The "if_analysis_name" parameter should be a string and here, it is set to None -> perhaps using the "None" string value instead of None on line 241 would be better ?

TickyWill replied:
Done. thanks

BaptisteRefalo commented on bmfuncts/update_impact_factors.py:
```diff
@@ -300,46 +344,105 @@ def _build_recent_year_if_df(institute, org_tup, bibliometer_path,
 
 
-def update_inst_if_database(institute, org_tup, bibliometer_path,
+def _set_if_files_param(institute, wf_path):
```
Docstring missing :)

TickyWill replied:
Docstrings completed. Thanks

BaptisteRefalo commented on bmgui/gui_globals.py:
There are 2 lines where you use the "assert" keyword (l.110 & l.112)
- I give you the message from Pylint (this keyword is normally only used for testing - unit tests...): "AST100: Asserts should only be used in tests. Asserts are typically bypassed in a production environment."

TickyWill replied:
Don't have found the use of this word....

BaptisteRefalo commented on bmgui/main_utils.py:
The docstrings should be more detailed (args & returns)

TickyWill replied:
Docstrings compleyed. Thanks

BaptisteRefalo commented on bmgui/pages_classes.py:
Same here :) -> The docstrings should be more detailed (args & returns)

TickyWill replied:
Docstrings compleyed. Thanks

BaptisteRefalo commented on bmgui/pages_utils.py:
Same also here for the functions' docstrings

TickyWill replied:
Docstrings compleyed. Thanks

BaptisteRefalo reviewed (commented):
PR review done :) Just a few elements to improve.

I also created the branch "refactoring-GUI-small-enhancements" based on the this branch ("refactoring-GUI") in which I already did some small modifications.
-> I'll create a PR to merge my branch with yours: this merge should be done before the one with the v6.2.0

Regarding the functional tests, I did not find the workspace folder for this version, so I could not test the app functionalities: these tests can be done later (and you surely have done some already ;) )

BaptisteRefalo committed:
Refactoring GUI complementary changes

Some enhancements based on my PR review

BaptisteRefalo committed:
rollback to corpuses

corpora -> corpuses (I added the word in my PyCharm English dictionary, it should note notify this word as incorrect in the future)

TickyWill committed:
Consolidated PR changes before merge into parent branch refactoring-GUI

TickyWill committed:
Consolidated PR changes before merge into v6.2.0

TickyWill committed:
Merge branch 'v6.2.0' into refactoring-GUI
